src/index.cpp

```diff
@@ -1486,11 +1493,13 @@ namespace diskann {
     ScratchStoreManager<InMemQueryScratch<T>> manager(_query_scratch);
     auto                                      scratch = manager.scratch_space();
 
-    if (L > scratch->search_l) {
-      scratch->resize_for_query(L);
-      diskann::cout << "Expanding query scratch_space. Was created with Lsize: "
-                    << scratch->search_l << " but search L is: " << L
-                    << std::endl;
+    if (L > scratch->get_L()) {
+      diskann::cout << "Attempting to expand query scratch_space. Was created "
+                    << "with Lsize: " << scratch->get_L()
+                    << " but search L is: " << L << std::endl;
+      scratch->resize_for_new_L(L);
+      diskann::cout << "Resize completed. Let scratch->L is "
+                    << scratch->get_L() << std::endl;
     }
 
     return search_impl(query, K, L, indices, distances, scratch);
```

Comment:
`search_impl` has two issues (don't remember if it was fixed in some other branch, so commenting just in case):

1. `auto best_L_nodes = scratch->best_l_nodes();` is missing a reference after `auto`.
2. The `EXEC_ENV_OLS` path refers to `it.distance`, where `it` variable doesn't exist.

Reply:
Yes, that is the other commit in a separate PR I was going to make to fix exactly this. Do you want it in this PR? I can add it here too